File: ui/src/types/videoMetadata.ts
import { type WebResponse } from "./webResponse"

export interface FrameInfo {
    Frame: number
    Keep: boolean
    Url: string
}

export interface VideoMetadata extends WebResponse {
    TotalSeconds: number
    Frames: FrameInfo[]
    LastSkip: number
    VideoSrc: string
}

File: app/types/videoMetadata.go
package types

type FrameInfo struct {
	Frame int
	Keep  bool
	Url   string
}

type VideoMetadata struct {
	Status       States
	Message      string
	TotalSeconds int
	LastSkip     int
	Frames       []FrameInfo
	VideoSrc     string
}

type VideoOutputOptions struct {
	OutputVolume int
}
